### DISCUSSION TITLE
How to use API with no UI

### DISCUSSION THREAD

**Discussion by ivan-diachenko-td** [vslavik/winsparkle#269] [Q&A] [answered]
Hello team,

Is it possible to use WinSparkle API completely without any UI? For example, I want to wrap your API in my own to look smth like this:

```c
checkForUpdate(callback onFound, callback onNotFound, callback onError);
```

I don't want my users to see any of WinSparkle UI. Is it technically possible?

I am looking forward to hearing from you.

**Comment by vslavik** [comment, marked as answer]
> Is it technically possible?

*Possible*, yes - by submitting a PR doing the work to change the API. That would be welcome.

Can you do it out of the box? Right now, no.
